backend/src/main/java/com/salat/briene/entities/ArticleState.java

```diff
@@ -1,5 +1,34 @@
 package com.salat.briene.entities;
 
+import com.salat.briene.exceptions.IllegalArticleStateException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
 public enum ArticleState {
-    ARTICLE_PUBLISHED, ARTICLE_IN_EDITING
+    PUBLISHED("published", "publish"),
+    IN_EDITING("drafts", "save"),
+    ALL("all", null);
+
+    private String description;
+
+    private String actionForThisState;
+
+    public static ArticleState getFromDescription(String description) throws IllegalArticleStateException {
+        return switch (description.toLowerCase()) {
```

Comment:
FYI: you can iterate through enum values and try find match, so you won't need to update this method on enum update. We can't forget to do it later, and it's also easier to read (we see match of  description - method argument and same enum value).